src/rulesets/Base/array.jl

```diff
@@ -37,6 +37,18 @@ function rrule(
     return Base.vect(X...), vect_pullback
 end
 
+# This rule isn't certain to be correct.
+# This rule _is_ correct when elements of `X` pass straight through into the output without
+# modification. e.g. if the output type of `vect` is a `Vector{Any}`, then this should be
+# correct. If, however, the type of some elements of `X` change in this operation in such a
+# way that their cotangents require projection on the reverse pass of AD, this projection
+# won't happen, possibly causing problems later on in the reverse pass.
+# Zygote and Diffractor fall over completely without this rule.
```

Comment:
```suggestion
# This should be fine for everything in the stdlibs though.
# Zygote and Diffractor fall over completely without this rule.
```
How about this?

Reply:
sure. I would even go further and say, that it is fine everythign that doesn't overload `vect` in a weird way.

I would also at this stage be willing to just remove the comment entirely, as I am convienced it will never cause a problem.
`vect` is not something anything should overload?

Reply:
Okay. I'll just remove the entire comment. If it causes problems, we can revisit.

Reply:
I've changed the comment. Please resolve if you're happy, and I'll go ahead and get this merged and tagged.